route-controller/cmd/main.go

```diff
@@ -0,0 +1,204 @@
+)
+
+func init() {
+	utilruntime.Must(buildv1alpha2.AddToScheme(scheme))
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(gatewayv1beta1.Install(scheme))
+	utilruntime.Must(korifiv1alpha1.AddToScheme(scheme))
+	//+kubebuilder:scaffold:scheme
+}
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -generate
+
+//counterfeiter:generate -o fake -fake-name Client sigs.k8s.io/controller-runtime/pkg/client.Client
+//counterfeiter:generate -o fake -fake-name EventRecorder k8s.io/client-go/tools/record.EventRecorder
+//counterfeiter:generate -o fake -fake-name StatusWriter sigs.k8s.io/controller-runtime/pkg/client.StatusWriter
+
+func main() {
+	var metricsAddr string
+	var enableLeaderElection bool
+	var probeAddr string
+
+	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
+	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
+	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
+		"Enable leader election for controller manager. "+
+			"Enabling this will ensure there is only one active controller manager.")
+
+	configPath, found := os.LookupEnv("CONTROLLERSCONFIG")
+	if !found {
+		panic("CONTROLLERSCONFIG must be set")
+	}
+
+	controllerConfig, err := config.LoadFromPath(configPath)
+	if err != nil {
+		errorMessage := fmt.Sprintf("Config could not be read: %v", err)
+		panic(errorMessage)
+	}
+
+	logger, atomicLevel, err := tools.NewZapLogger(controllerConfig.LogLevel)
+	if err != nil {
+		panic(fmt.Sprintf("error creating new zap logger: %v", err))
+	}
+
+	ctrl.SetLogger(logger)
+
+	log.SetOutput(&tools.LogrWriter{Logger: ctrl.Log, Message: "HTTP server error"})
+
+	ctrl.Log.Info("starting Korifi controllers", "version", version.Version)
+
+	conf := ctrl.GetConfigOrDie()
+
+	mgr, err := ctrl.NewManager(conf, ctrl.Options{
+		Scheme: scheme,
+		WebhookServer: webhook.NewServer(webhook.Options{
+			Port: 9443,
+		}),
+		Metrics: metricsserver.Options{
+			BindAddress: metricsAddr,
+		},
+		HealthProbeBindAddress: probeAddr,
+		LeaderElection:         enableLeaderElection,
+		LeaderElectionID:       "13c200ec.cloudfoundry.org",
+	})
+	if err != nil {
+		setupLog.Error(err, "unable to initialize manager")
+		os.Exit(1)
+	}
+	controllersClient := k8s.IgnoreEmptyPatches(mgr.GetClient())
+	if os.Getenv("ENABLE_CONTROLLERS") != "false" {
+		controllersLog := ctrl.Log.WithName("controllers")
+
+		if !controllerConfig.DisableRouteController {
+			if err = routeControllers.NewReconciler(
+				controllersClient,
+				mgr.GetScheme(),
+				controllersLog,
+				controllerConfig,
+			).SetupWithManager(mgr); err != nil {
+				setupLog.Error(err, "unable to create controller", "controller", "CFRoute")
+				os.Exit(1)
+			}
+		}
+	}
+	// Setup webhooks with manager
+	if os.Getenv("ENABLE_WEBHOOKS") != "false" {
+
+		(&appswebhook.AppRevWebhook{}).SetupWebhookWithManager(mgr)
```

Comment:
The `AppRevWebhook` is related to CFApps and CFApps are not networker responsibility. No need to setup this webhook here